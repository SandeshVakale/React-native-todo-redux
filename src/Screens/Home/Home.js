import React from 'react';
import {
  ScrollView,
  View,
  TextInput,
  TouchableOpacity,
  SafeAreaView,
} from 'react-native';
import {useSelector} from 'react-redux';
import {v4 as uuidv4} from 'uuid';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {store} from '../../Store';
import {faPlusCircle} from '@fortawesome/free-solid-svg-icons';
import {Colors} from '../../Themes';
import Styles from './Styles';
import TodoCard from '../../Components/TodoCard';

const Home = () => {
  const {todo} = useSelector(state => state.todoModel);
  const [value, onChangeText] = React.useState('');

  return (
    <SafeAreaView style={Styles.container}>
      <ScrollView style={Styles.container}>
        {todo.map(item => (
          <TodoCard key={item.id} item={item} />
        ))}
      </ScrollView>
      <View style={Styles.inputContainer}>
        <TextInput
          style={Styles.input}
          onChangeText={text => onChangeText(text)}
          value={value}
          placeholder="Insert your todo!"
        />
        <TouchableOpacity
          onPress={() => {
            value !== '' &&
              store.dispatch({
                type: 'todo/addToTodo',
                payload: {name: value, id: uuidv4()},
              });
            onChangeText('');
          }}>
          <FontAwesomeIcon
            icon={faPlusCircle}
            size={32}
            color={Colors.background}
          />
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

export default Home;
