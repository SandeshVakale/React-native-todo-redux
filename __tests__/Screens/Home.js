import 'react-native';
import React from 'react';
import Home from '../../src/Screens/Home/Home';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

const mockStore = configureStore([]);

test('Home screen renders correctly', () => {
  const initialStore = {
    todoModel: {todo: []},
  };
  let store = mockStore(initialStore);
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
