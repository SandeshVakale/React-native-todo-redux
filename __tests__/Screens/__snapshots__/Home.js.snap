// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "position": "absolute",
        "zIndex": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Insert your todo!"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 10,
          "borderWidth": 1,
          "padding": 10,
          "width": "85%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <
        color="rgba(52, 52, 52, 0.8)"
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f055",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
            ],
            "iconName": "plus-circle",
            "prefix": "fas",
          }
        }
        size={32}
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
